### PR TITLE
[Audit Fix] - Added tests for EIP 712 attack

### PR DESCRIPTION
**Pull request by sanandnarayan**
Added tests for EIP 712 attack

### CODE REVIEW

File: packages/hardhat/test/integration-tests/crabv2/crab-hedging-otc.ts [hunk 1]
```diff
@@ -1000,6 +1000,39 @@ describe("Crab V2 flashswap integration test: time based hedging", function () {
...
+            await wSqueeth.connect(trader).approve(crabStrategyV2.address, toGet);
+
+            const orderHash = {
+                bidId: 0,
+                trader: trader.address,
+                quantity: toSell,
+                price: 1,
+                isBuying: false,
+                expiry: (await provider.getBlock(await provider.getBlockNumber())).timestamp + 600,
+                nonce: await crabStrategyV2.nonces(trader.address),
```
Comment (alpinechicken) [inline packages/hardhat/test/integration-tests/crabv2/crab-hedging-otc.ts:1022]: gonna need a new test when nonce pr merged

**Review by alpinechicken — APPROVED** [review]
lgtm

File: packages/hardhat/test/integration-tests/crabv2/crab-hedging-otc.ts (hunk 1 above)
Comment (sanandnarayan) [reply packages/hardhat/test/integration-tests/crabv2/crab-hedging-otc.ts:1022]: nonce PR might not get merged depending on the feedback from MMs

**Review by aleone — APPROVED** [review]

**Comment by sanandnarayan** [comment]
ran a linter.

**Pull request merged**